### PR TITLE
feat(prejoin) Add possibility to hide extra join options buttons

### PR DESCRIPTION
**Pull request by horymury**
- fix line height issue on extra join options buttons
- small refactor

### CODE REVIEW

File: react/features/prejoin/components/DropdownButton.js [hunk 1]
```diff
@@ -59,7 +59,7 @@ const styles = theme => {
             display: 'flex',
             height: 40,
             fontSize: 15,
-            lineHeight: 24,
+            lineHeight: '24px',
```
Comment (saghul) [inline react/features/prejoin/components/DropdownButton.js:62]: why do we use px here but not for the height for example?

File: react/features/prejoin/functions.js [hunk 2]
```diff
@@ -158,7 +158,7 @@ export function isJoinByPhoneDialogVisible(state: Object): boolean {
  */
 export function isPrejoinPageEnabled(state: Object): boolean {
     return navigator.product !== 'ReactNative'
-        && state['features/base/config'].prejoinPageEnabled
+        && state['features/base/config'].prejoinConfig.enabled
```
Comment (saghul) [inline react/features/prejoin/functions.js:161]:
```suggestion
        && state['features/base/config'].prejoinConfig?.enabled
```

File: react/features/settings/functions.js [hunk 3]
```diff
@@ -125,7 +125,7 @@ export function getMoreTabProps(stateful: Object | Function) {
         languages: LANGUAGES,
         showLanguageSettings: configuredTabs.includes('language'),
         showModeratorSettings,
-        showPrejoinSettings: state['features/base/config'].prejoinPageEnabled,
+        showPrejoinSettings: state['features/base/config'].prejoinConfig.enabled,
```
Comment (saghul) [inline react/features/settings/functions.js:128]:
```suggestion
        showPrejoinSettings: state['features/base/config'].prejoinConfig?.enabled,
```

File: react/features/prejoin/components/Prejoin.js [hunk 4]
```diff
@@ -290,15 +330,25 @@ class Prejoin extends Component<Props, State> {
...
             showDialog,
             t,
             videoTrack
         } = this.props;
+        const { _closeDialog, _onDropdownClose, _onJoinButtonClick, _onJoinKeyPress,
+            _onOptionsClick, _setName } = this;
+
+        const extraJoinButtons = this._getExtraJoinButtons();
+        let extraButtonsToRender = Object.values(extraJoinButtons).filter(val =>
+            !prejoinConfig.hideExtraJoinButtons?.includes(val.key)
```
Comment (saghul) [inline react/features/prejoin/components/Prejoin.js]:
```suggestion
            !prejoinConfig?.hideExtraJoinButtons?.includes(val.key)
```

File: react/features/prejoin/components/Prejoin.js [hunk 5]
```diff
@@ -290,15 +330,25 @@ class Prejoin extends Component<Props, State> {
...
+        );
 
-        const { _closeDialog, _onDropdownClose, _onJoinButtonClick, _onJoinKeyPress, _showDialogKeyPress,
-            _onJoinConferenceWithoutAudioKeyPress, _onOptionsClick, _setName, _showDialog } = this;
+        if (!hasJoinByPhoneButton) {
+            extraButtonsToRender = extraButtonsToRender.filter(btn => btn.key !== 'by-phone');
+        }
+        const hasExtraJoinButtons = Boolean(extraButtonsToRender && extraButtonsToRender.length);
```
Comment (saghul) [inline react/features/prejoin/components/Prejoin.js:351]: I don't see how `extraButtonsToRender` would be undefined here.

**Review by saghul — COMMENTED** [review]
Left some comments PTAL.

File: react/features/prejoin/functions.js (hunk 2 above)
Comment (horymury) [reply react/features/prejoin/functions.js:161]: `prejoinConfig` is always set, at least as an empty object , in the config reducer

File: react/features/settings/functions.js (hunk 3 above)
Comment (horymury) [reply react/features/settings/functions.js:128]: -||-

File: react/features/prejoin/components/Prejoin.js (hunk 4 above)
Comment (horymury) [reply react/features/prejoin/components/Prejoin.js]: -||-

File: react/features/prejoin/components/Prejoin.js (hunk 5 above)
Comment (horymury) [reply react/features/prejoin/components/Prejoin.js:351]: true, I'll remove that check

File: react/features/prejoin/components/DropdownButton.js (hunk 1 above)
Comment (horymury) [reply react/features/prejoin/components/DropdownButton.js:62]: mUI quirk, for lineHeight it takes em as default. This was causing a pretty ugly bug on the dropdown as the line height was over 300px for the buttons labels.

File: react/features/prejoin/functions.js (hunk 2 above)
Comment (saghul) [reply react/features/prejoin/functions.js:161]: Ah true.

Comment (saghul) [reply react/features/prejoin/functions.js:161]: Wait. Yes it is... for now. 6 months from now, when we remove the backwards compat code it won't. Please take this into account.

File: react/features/prejoin/components/DropdownButton.js (hunk 1 above)
Comment (saghul) [reply react/features/prejoin/components/DropdownButton.js:62]: All I hear is some CSS gibberish, so I trust you :-)

File: react/features/prejoin/functions.js (hunk 2 above)
Comment (horymury) [reply react/features/prejoin/functions.js:161]: good point, I'll add the check.

File: config.js [hunk 6]
```diff
@@ -496,6 +496,14 @@ var config = {
...
 
+    // Configs for prejoin page.
+    // prejoinConfig: {
+    //     // Replaces prejoinPageEnabled config.
+    //     enabled: true,
+    //     // List of buttons to hide from the extra join options dropdown.
+    //     hideExtraJoinButtons: ['no-audio', 'by-phone']
+    // },
+
     // When 'true', it shows an intermediate page before joining, where the user can configure their devices.
```
Comment (saghul) [inline config.js:507]: Can you remove this entry?

**Review by saghul — APPROVED** [review]
LGTM with a nit.

File: config.js (hunk 6 above)
Comment (horymury) [reply config.js]: done. also moved the comment on the 'enabled' flag instead.

**Comment by saghul** [comment]
Jenkins please test this please.

**Review by saghul — APPROVED** [review]

**Pull request merged**